test/common/shared_pool/shared_pool_test.cc

```diff
@@ -142,7 +142,7 @@ TEST_F(SharedPoolTest, GetObjectAndDeleteObjectRaceForSameHashValue) {
   // Because the storage is actually a new weak_ptr and the reference count is not zero, it is not
   // deleted
   dispatcher_->post([&pool, this]() {
-    EXPECT_EQ(1, pool->poolSize());
+    EXPECT_EQ(0, pool->poolSize());
```

Comment:
Investigate test failure.

Reply:
This fails because `absl::flat_hash_set<T, Hash, Compare>` will skip a call to `set.emplace(x)` if it finds a `T` which returns true for `Compare`.